### ISSUE TITLE
Possible undefined behavior

### ISSUE THREAD

**Issue by a-day-old-bagel** [alexsaalberg/game_of_stones#1]
Running in linux, I *sometimes* get a segfault and sometimes I don't. It seems to always happen in the first frame if it happens at all, and my debugger reports it as coming from the glDrawElements call in Shape::draw when it does happen.

**Comment by alexsaalberg** [comment]
I've never tried running it in linux, and the only time I tried running it on another OSX computer it crashed because it didn't have the right dependencies (which should probably be bundled with it or written down somewhere).

So you're saying that sometimes it _does_ render correctly? (Like a green hill or something)

**Comment by a-day-old-bagel** [comment]
Yes, to get it to run in Linux all I had to do was put the glm and PolyVox stuff in the include path (which amounted to installing those via my package manager) and fixing some of the bullet3 include directives. It worked pretty much out of the box that way, which is why I said on your reddit thread (YonderBagel) that the cross-platformness wasn't really challenging.
So the times that it does work, it appears to work like in your video of it and everything. I did have to add a multiplier to the mouse speed though, as a side note, since I think you must have developed it with a very high DPI setting on your mouse lol.

**Comment by alexsaalberg** [comment]
I'm glad to know that it works somewhat on linux! Yes, I probably do have a high DPI lol.
I'll look into this issue as well as making it easier to build the project (possibly by including bullet and glm).

I copied the project into a new repo so I'll move this there, too
